forest/cli/src/cli/info_cmd.rs

```diff
@@ -0,0 +1,406 @@
+        let base_fee_fmt = self.base_fee.pretty();
+        let behind = format!("{}", humantime::format_duration(self.behind));
+        format!(
+            "[sync: {}! ({} behind)] [basefee: {base_fee_fmt}] [epoch: {}]",
+            self.sync_status, behind, self.epoch
+        )
+    }
+}
+
+impl From<NodeStatusInfo> for NodeInfoOutput {
+    fn from(node_status_info: NodeStatusInfo) -> NodeInfoOutput {
+        let health = node_status_info.health;
+        let uptime = (Utc::now() - node_status_info.start_time)
+            .to_std()
+            .expect("failed converting to std duration");
+        let fmt_uptime = fmt_duration(uptime);
+        let uptime = format!(
+            "{fmt_uptime} (Started at: {})",
+            node_status_info
+                .start_time
+                .with_timezone(&chrono::offset::Local)
+        )
+        .normal();
+
+        let chain_status = node_status_info.chain_status().blue();
+        let network = node_status_info.network.green();
+        let wallet_address = node_status_info
+            .default_wallet_address
+            .clone()
+            .unwrap_or("address not set".to_string())
+            .bold();
+        let health = {
+            let h = format!("{health:.2}%\n\n");
+            if health > 85. {
+                h.green()
+            } else if health > 50. {
+                h.yellow()
+            } else {
+                h.red()
+            }
+        };
+
+        let wallet_balance = match balance(node_status_info.default_wallet_address_balance) {
+            Ok(bal) => format!("[balance: {}]", bal).bold(),
+            Err(_) => "".bold(),
+        };
+
+        NodeInfoOutput {
+            chain_status,
+            network,
+            uptime,
+            health,
+            wallet_address,
+            wallet_balance,
+        }
+    }
+}
+
+impl NodeInfoOutput {
+    fn set_color(mut self, color: &LoggingColor) -> Self {
+        if !color.coloring_enabled() {
+            self.chain_status = self.chain_status.clear();
+            self.network = self.network.clear();
+            self.uptime = self.uptime.clear();
+            self.health = self.health.clear();
+            self.wallet_address = self.wallet_address.clear();
+            self.wallet_balance = self.wallet_balance.clear();
+        }
+
+        self
+    }
+}
+
+impl InfoCommand {
+    pub async fn run(&self, config: Config, opts: &CliOpts) -> anyhow::Result<()> {
+        let node_status = node_status((), &config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let head = chain_head(&config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let network = chain_get_name((), &config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
+
+        let start_time = start_time(&config.client.rpc_token)
+            .await
+            .map_err(handle_rpc_err)?;
```

Comment:
This is ugly, can you refactor using `try_join` please

Reply:
neat, thx.